proto/caraml/upi/v1/observation_log.proto

```diff
@@ -46,3 +47,105 @@ message LogObservationsRequest {
 message LogObservationsResponse {
     string observation_batch_id = 1;
 }
+
+// Data source where Observation logs should be retrieved from
+enum ObservationServiceDataSourceType {
+  OBSERVATION_SERVICE_DATA_SOURCE_TYPE_UNSPECIFIED = 0;
+  // No-Op represents no need to fetch logs from any data source, this should be selected if
+  // Observation Service should be deployed for just the eager API
+  OBSERVATION_SERVICE_DATA_SOURCE_TYPE_NOOP = 1;
+  // Observation Service will poll logs from a Kafka source
+  OBSERVATION_SERVICE_DATA_SOURCE_TYPE_KAFKA = 2;
+}
+
+// Configurations of Data source where Observation logs should be retrieved from
+message ObservationServiceDataSource {
+  ObservationServiceDataSourceType type = 1;
+  KafkaConfig kafka_config = 2;
+}
+
+// Data sink where Observation logs would be flushed to
+enum ObservationServiceDataSinkType {
+  OBSERVATION_SERVICE_DATA_SINK_TYPE_UNSPECIFIED = 0;
+  // No-Op represents no need to flush logs to any data sink
+  OBSERVATION_SERVICE_DATA_SINK_TYPE_NOOP = 1;
+  // Observation Service will publish logs to standard output
+  OBSERVATION_SERVICE_DATA_SINK_TYPE_STDOUT = 2;
+  // Observation Service will flush logs to a Kafka sink
+  OBSERVATION_SERVICE_DATA_SINK_TYPE_KAFKA = 3;
+  // Observation Service will flush logs to Fluentd
+  OBSERVATION_SERVICE_DATA_SINK_TYPE_FLUENTD = 4;
+}
+
+// Configurations of Data sink where Observation logs would be flushed to
+message ObservationServiceDataSink {
+  ObservationServiceDataSinkType type = 1;
+  KafkaConfig kafka_config = 2;
+  FluentdConfig fluentd_config = 3;
+}
+
+// ObservationService describes details of a Observation Service
+message ObservationServiceConfig {
+  // Unique identifier of an Observation Service deployed by Dataset Service.
+  string id = 1;
+  // Configuration for pull-based Observation Service data source.
+  ObservationServiceDataSource source = 2;
+  // Configuration for Observation Service data sink.
+  ObservationServiceDataSink sink = 3;
+}
+
+// Request message for ListObservationServices method.
+message ListObservationServicesRequest {
```

Comment:
Can we put all protos related to the `DatasetService` contract (anything that ends with `Request` and `Response`) in the same file as its service definitions( i.e. in [dataset_service.proto](https://github.com/caraml-dev/universal-prediction-interface/pull/20/files#diff-9bf0bd9f96ed0510e74f1f14bf226d552628177722eb645e238fa5481a7f95fe))? 

The intent is to store all service definition in one proto file and all other entity definition (like Log, LogWriter) in other proto file.